src/cpp/fastdds/xtypes/dynamic_types/idl_parser/IdlParser.hpp

```diff
@@ -0,0 +1,1725 @@
+    }
+
+private:
+
+    friend class Context;
+    //using LabelsCaseMemberPair = std::pair<std::vector<std::string>, Member>;
+
+    //peg::parser parser_;
+    Context* context_;
+
+    traits<DynamicType>::ref_type type_spec(
+            std::map<std::string, std::string>& state,
+            const std::string& type)
+    {
+        DynamicTypeBuilderFactory::_ref_type factory {DynamicTypeBuilderFactory::get_instance()};
+        DynamicTypeBuilder::_ref_type builder;
+        DynamicType::_ref_type xtype;
+
+        if (type == "boolean")
+        {
+            xtype = factory->get_primitive_type(TK_BOOLEAN);
+        }
+        else if (type == "int8")
+        {
+            xtype = factory->get_primitive_type(TK_INT8);
+        }
+        else if (type == "uint8")
+        {
+            xtype = factory->get_primitive_type(TK_UINT8);
+        }
+        else if (type == "int16")
+        {
+            xtype = factory->get_primitive_type(TK_INT16);
+        }
+        else if (type == "uint16")
+        {
+            xtype = factory->get_primitive_type(TK_UINT16);
+        }
+        else if (type == "int32")
+        {
+            xtype = factory->get_primitive_type(TK_INT32);
+        }
+        else if (type == "uint32")
+        {
+            xtype = factory->get_primitive_type(TK_UINT32);
+        }
+        else if (type == "int64")
+        {
+            xtype = factory->get_primitive_type(TK_INT64);
+        }
+        else if (type == "uint64")
+        {
+            xtype = factory->get_primitive_type(TK_UINT64);
+        }
+        else if (type == "float")
+        {
+            xtype = factory->get_primitive_type(TK_FLOAT32);
+        }
+        else if (type == "double")
+        {
+            xtype = factory->get_primitive_type(TK_FLOAT64);
+        }
+        else if (type == "long double")
+        {
+            xtype = factory->get_primitive_type(TK_FLOAT128);
+        }
+        else if (type == "char")
+        {
+            xtype = factory->get_primitive_type(TK_CHAR8);
+        }
+        else if (type == "wchar" || type == "char16")
+        {
+            xtype = factory->get_primitive_type(TK_CHAR16);
+        }
+        else if (type == "string")
+        {
+            TypeDescriptor::_ref_type type_descriptor {traits<TypeDescriptor>::make_shared()};
+            type_descriptor->kind(TK_STRING8);
+            type_descriptor->element_type(factory->get_primitive_type(TK_CHAR8));
+
+            if (state.count("string_size"))
+            {
+                uint32_t size = std::atoi(state["string_size"].c_str());
+                type_descriptor->bound({size});
+                state.erase("string_size");
+            }
+            builder = factory->create_type(type_descriptor);
+            xtype = builder->build();
+        }
+        else if (type == "wstring")
+        {
+            TypeDescriptor::_ref_type type_descriptor {traits<TypeDescriptor>::make_shared()};
```

Comment:
You can use `DynamicTypeBuilderFactory::create_wstring_type`

Reply:
Done.